subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/RequiresExtension.groovy

```diff
@@ -79,8 +79,13 @@ class RequiresExtension implements IAnnotationDrivenExtension<Requires> {
         }
 
         PredicatesFile.checkValidCombinations(predicateClassNames, acceptedCombinations)
-        // If all preconditions are met, we DON'T skip the tests
-        specOfFeature.skipped |= !TestPrecondition.allSatisfied(annotation.value())
+
+        for (Class<? extends TestPrecondition> preconditionClass : annotation.value()) {
+            if (!TestPrecondition.satisfied(preconditionClass)) {
+                specOfFeature.skip("Precondition not satisfied: " + preconditionClass.simpleName)
```

Comment:
💭 I wonder if it would make sense to summarize all of the failed preconditions rather than just the first one?  In other words, build up a string of failed preconditions and provide one skipped string listing all of them. 

It would also be nice if `TestPrecondition` implementations provided a friendly description that could be used here, but there are a lot of them that would need to be refactored.  Maybe not worth the effort.